.github/workflows/setup-gcloud-it.yml

```diff
@@ -177,3 +177,81 @@ jobs:
       with:
         entrypoint: '/bin/sh'
         args: '-euc "test -n "${GOOGLE_APPLICATION_CREDENTIALS}" && test -r "${GOOGLE_APPLICATION_CREDENTIALS}"'
+
+  wif:
+    if: ${{ github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name && github.actor != 'dependabot[bot]' }}
+    name: 'with wif'
+    runs-on: '${{ matrix.os }}'
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+        - 'ubuntu-latest'
+        - 'windows-latest'
+        - 'macos-latest'
+
+    permissions:
+      id-token: 'write'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: 'actions/setup-node@v2'
+        with:
+          node-version: '12.x'
+
+      - id: build
+        name: Build dist
+        run: |-
+          npm ci
+          npm run build
+
+      - uses: google-github-actions/auth@main
+        with:
+          workload_identity_provider: ${{ secrets.WIF_PROVIDER_NAME }}
+          service_account: ${{ secrets.OIDC_AUTH_SA_EMAIL }}
+
+      - name: 'setup-gcloud'
+        uses: './'
+      
+      - id: 'gcloud'
+        shell: 'bash'
+        run: |-
+          gcloud secrets versions access "latest" --secret "${{ secrets.OIDC_AUTH_TEST_SECRET_NAME }}"
+
+  auth-key:
+    if: ${{ github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name && github.actor != 'dependabot[bot]' }}
```

Comment:
To consider in a follow-up. I wonder if we can store this conditional as an [env](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#env) and then just reference the `env` in the conditional instead. I think:

```yaml
if: ${{ env.CAN_ACCESS_SECRETS }}
```

might be a little clearer. Alternatively, maybe we could do something with [defaults](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#defaults) to have a default `if`?

Low priority, just thinking out loud.

Reply:
Yeah its getting cumbersome. Opened https://github.com/google-github-actions/setup-gcloud/issues/414 to track for followup.